src/main/java/com/yelp/nrtsearch/server/config/ScriptCacheConfig.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.config;
+
+import java.util.concurrent.TimeUnit;
+
+/** Configuration class for script cache. */
+public class ScriptCacheConfig {
+  static final int DEFAULT_CONCURRENCY_LEVEL = 4;
+  static final int DEFAULT_SIZE = 1000;
+  static final long DEFAULT_EXPIRATION_TIME = 1;
+  static final String DEFAULT_TIME_UNIT = "DAYS";
+
+  private final int concurrencyLevel;
+  private final int maximumSize;
+  private final long expirationTime;
+  private final TimeUnit timeUnit;
+
+  /**
+   * Create instance from provided configuration reader.
+   *
+   * @param configReader config reader
+   * @return class instance
+   */
+  public static ScriptCacheConfig fromConfig(YamlConfigReader configReader) {
+    int concurrencyLevel =
+        configReader.getInteger("ScriptCacheConfig.concurrencyLevel", DEFAULT_CONCURRENCY_LEVEL);
+    int maximumSize = configReader.getInteger("ScriptCacheConfig.maximumSize", DEFAULT_SIZE);
+    long expirationTime =
+        configReader.getLong("ScriptCacheConfig.expirationTime", DEFAULT_EXPIRATION_TIME);
+    TimeUnit timeUnit =
+        getTimeUnitFromString(
+            configReader.getString("ScriptCacheConfig.timeUnit", DEFAULT_TIME_UNIT));
+    return new ScriptCacheConfig(concurrencyLevel, maximumSize, expirationTime, timeUnit);
+  }
+
+  /**
+   * @param timeUnitStr
+   * @return expiration time unit from config
+   * @throws IllegalArgumentException
+   */
+  public static TimeUnit getTimeUnitFromString(String timeUnitStr) throws IllegalArgumentException {
+    if (timeUnitStr == null) {
+      throw new IllegalArgumentException("script cache expiration time unit cannot be null");
+    }
```

Comment:
nit: a cleaner way to do this would be using `Objects.requireNonNull()`, which will throw a `NullPointerException`. https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-
Also, since these are `RuntimeException`s, they don't need to be declared on the method signature.